tests/unit/amo/sagas/test_collections.js

```diff
@@ -559,22 +559,28 @@ describe(__filename, () => {
         ).not.toContain(unloadCollectionBySlug(collectionSlug).type);
       });
 
-      it('redirects to the existing slug after update', async () => {
-        mockApi.expects('updateCollection').returns(Promise.resolve());
+      it('redirects to the resulting slug after update', async () => {
+        const submittedSlug = 'submitted-slug';
+        const returnedSlug = 'returned-slug';
+
+        const collectionDetailResponse = createFakeCollectionDetail({
+          slug: returnedSlug,
+        });
+
+        mockApi.expects('updateCollection').returns(collectionDetailResponse);
```

Comment:
This should return a promise:
```js
        mockApi
          .expects('updateCollection')
          .returns(Promise.resolve(collectionDetailResponse));
```
I think it only works as is because sinon stubs are a bit magical and they have a fallback promise implementation (this is just a guess). Relying on magic makes me nervous.